runtime/lua/vim/treesitter.lua

```diff
@@ -438,11 +426,10 @@ end
 ---@param lang (string|nil) Language of the parser (default: from buffer filetype)
 function M.start(bufnr, lang)
   bufnr = bufnr or api.nvim_get_current_buf()
-  local parser = M._get_parser(bufnr, lang)
-  if not parser then
-    vim.notify('No parser for the given buffer.', vim.log.levels.WARN)
-    return
-  end
+  local parser = assert(
+    M.get_parser(bufnr, lang, { error = false }),
+    string.format('Highlighting could not be started for buffer: %s. No parser found.', bufnr)
```

Comment:
Sorry to "one more thing" you, but I think the error message should contain as much context as possible -- in this case also the language name. (This may be called from "distros", so the context may not be always clear to the user.)

(This is related to a regression on treesitter highlight error messages, which used to contain the language and query name.)

Reply:
Let me know if this change is not acceptable; I think the behavior is slightly less intuitive now because often `start()` is passed `nil` and the language is determined in the `get_parser` side, but now we shoot out the error message in `start()` (before that `lang` calculation) so the error message may be less helpful in a lot of cases (i.e. cases where you are *not* passing the language with something like `vim.treesitter.start(0, 'javascript')`